include/device.h

```diff
@@ -427,6 +427,24 @@ device_from_handle(device_handle_t dev_handle)
 	return dev;
 }
 
+/**
+ * @brief Prototype for functions used when iterating over a set of devices.
+ *
+ * Such a function may be used in API that identifies a set of devices and
+ * provides an visitor API supporting caller-specific interaction with each
+ * device in the set.
+ *
+ * The visit is said to succeed if the visitor returns a non-negative value.
+ *
+ * @param dev a device in the set being iterated
+ *
+ * @param context state used to support the visitor function
+ *
+ * @return A non-negative number to allow walking to continue, and a negative
+ * error code to case the iteration to stop.
+ */
+typedef int (*device_visitor_cb)(const struct device *dev, void *context);
```

Comment:
Looking through the other callback typedefs in the driver APIs, most of them seem to be named `foo_callback_t`.

Did you consider that and reject it? If so, why?

Reply:
`device_visitor_callback_t` seemed a little long.

`git grep -P '_cb\b'| grep ':typedef'` shows 26 with `_cb`, 35 with `_callback_t`, though I agree device headers tend toward the latter

Changed.